backend/app/services/llm/providers/gai.py

```
@@ -324,16 +377,92 @@ def _execute_tts(
+            contents=contents,
+            config=GenerateContentConfig(**generation_kwargs),
+        )
+
+        if response.usage_metadata:
+            input_tokens = response.usage_metadata.prompt_token_count or 0
+            output_tokens = response.usage_metadata.candidates_token_count or 0
+            total_tokens = response.usage_metadata.total_token_count or 0
+            reasoning_tokens = response.usage_metadata.thoughts_token_count or 0
+        else:
+            logger.warning(
+                f"[GoogleAIProvider._execute_text] Response missing usage_metadata, using zeros"
```

Ruff (0.15.2)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)